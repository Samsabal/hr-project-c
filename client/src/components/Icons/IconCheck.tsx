import React from 'react'
import { IconProps } from '../../types/IconProps'

export const IconCheck = ({ size, fill, color, strokeWidth = 2 }: IconProps) => {
    return (
        <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill="none">
            <path d="M20 6L9 17L4 12" strokeWidth={strokeWidth} strokeLinecap="round" strokeLinejoin="round" />
        </svg>
    )
}
