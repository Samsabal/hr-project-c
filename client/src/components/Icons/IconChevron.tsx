import React from 'react'

import { IconProps } from '../../types/IconProps'

export const IconChevron = ({ size, fill, color, strokeWidth = 2, direction }: IconProps) => {
    switch (direction) {
        case 'down':
            return (
                <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill="none">
                    <path d="M6 9L12 15L18 9" strokeWidth={strokeWidth} strokeLinecap="round" strokeLinejoin="round" />
                </svg>
            );
        case 'up':
            return (
                <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill="none">
                    <path d="M18 15L12 9L6 15" strokeWidth={strokeWidth} strokeLinecap="round" strokeLinejoin="round" />
                </svg>
            );
    }
    // Return a chevron pointing right by default
    return (
        <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill="none">
            <path d="M9 18L15 12L9 6" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" />
        </svg>
    );
}