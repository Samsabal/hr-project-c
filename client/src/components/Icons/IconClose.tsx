import React from 'react'
import { IconProps } from '../../types/IconProps'

export const IconClose = ({ size, fill, color, strokeWidth = 2 }: IconProps) => {
    return (
        <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill="none">
            <path d="M18 6L6 18M6 6L18 18" strokeWidth={strokeWidth} strokeLinecap="round" strokeLinejoin="round" />
        </svg>
    )
}
