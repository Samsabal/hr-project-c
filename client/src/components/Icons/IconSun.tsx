import React from 'react'
import { IconProps } from '../../types/IconProps'

export const IconSun = ({ size, fill = 'none', color, strokeWidth = 2 }: IconProps) => {
    return (
        <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill="none">
            <path opacity="0.12" d="M12 17C14.7614 17 17 14.7614 17 12C17 9.23858 14.7614 7 12 7C9.23858 7 7 9.23858 7 12C7 14.7614 9.23858 17 12 17Z" className={fill} />
            <path d="M12 2V4M12 20V22M4 12H2M6.31412 6.31412L4.8999 4.8999M17.6859 6.31412L19.1001 4.8999M6.31412 17.69L4.8999 19.1042M17.6859 17.69L19.1001 19.1042M22 12H20M17 12C17 14.7614 14.7614 17 12 17C9.23858 17 7 14.7614 7 12C7 9.23858 9.23858 7 12 7C14.7614 7 17 9.23858 17 12Z" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" />
        </svg>
    )
}
