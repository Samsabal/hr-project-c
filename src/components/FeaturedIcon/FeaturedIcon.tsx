import React from 'react'

export enum FeaturedIconSize {
    SM,
    MD,
    LG,
    XL
}

export enum FeaturedIconType {
    PRIMARY,
    GRAY,
    ERROR,
    SUCCESS
}

type FeaturedIconProps = {
    size: FeaturedIconSize,
    type: FeaturedIconType,
    icon: JSX.Element
}

export function FeaturedIcon({ size, type, icon }: FeaturedIconProps) {
    let featuredIconSize = "";
    let featuredIconType = "";

    switch (size) {
        case FeaturedIconSize.XL:
            featuredIconSize = "p-3 border-8";
            break;
        case FeaturedIconSize.LG:
            featuredIconSize = "p-2.5 border-8";
            break;
        case FeaturedIconSize.MD:
            featuredIconSize = "p-2 border-8";
            break;
        case FeaturedIconSize.SM:
            featuredIconSize = "p-1.5 border-8";
            break;
    }

    switch (type) {
        case FeaturedIconType.PRIMARY:
            featuredIconType = "bg-primary-100 border-primary-50 dark:border-dark-700 dark:bg-dark-600";
            break;
        case FeaturedIconType.GRAY:
            featuredIconType = "bg-gray-100 border-gray-50 dark:border-dark-700 dark:bg-dark-600";
            break;
        case FeaturedIconType.ERROR:
            featuredIconType = "bg-error-100 border-error-50 dark:border-dark-700 dark:bg-dark-600";
            break;
        case FeaturedIconType.SUCCESS:
            featuredIconType = "bg-success-100 border-success-50 dark:border-dark-700 dark:bg-dark-600";
            break;
    };

    return (
        <div className={`${featuredIconSize} ${featuredIconType}
            flex items-center justify-center rounded-full box-border`}>
            {icon}
        </div>
    )
}
