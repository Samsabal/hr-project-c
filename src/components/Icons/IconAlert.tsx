import React from 'react'
import { IconProps } from '../../types/IconProps'

export const IconAlert = ({ size, fill, color, strokeWidth = 2 }: IconProps) => {
    return (
        <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill={"none"}>
            <path className={fill} opacity="0.12" d="M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22Z" />
            <path d="M12 8V12M12 16H12.01M22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C17.5228 2 22 6.47715 22 12Z" strokeWidth={strokeWidth} strokeLinecap="round" strokeLinejoin="round" />
        </svg>

    )
}
