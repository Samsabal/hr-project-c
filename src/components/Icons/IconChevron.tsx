import React from 'react'

import { IconProps } from '../../types/IconProps'

export const IconChevron = ({ size, fill, color, strokeWidth = 2, direction }: IconProps) => {
    switch (direction) {
        case 'right':
            return (
                <svg className={color} width={size}height={size} viewBox="0 0 24 24" fill="none">
                    <path d="M9 18L15 12L9 6" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" />
                </svg>
            );
    }
    // Return a chevron pointing right by default
    return (
        <svg className={color} width={size}height={size} viewBox="0 0 24 24" fill="none">
            <path d="M9 18L15 12L9 6" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" />
        </svg>
    );
}