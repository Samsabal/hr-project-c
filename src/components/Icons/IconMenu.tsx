import React from 'react'
import { IconProps } from '../../types/IconProps'

export const IconMenu = ({ size, fill, color, strokeWidth = 2 }: IconProps) => {
    return (
        <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill="none">
            <path d="M3 12H21M3 6H21M9 18H21" strokeWidth={strokeWidth} strokeLinecap="round" strokeLinejoin="round" />
        </svg>
    )
}