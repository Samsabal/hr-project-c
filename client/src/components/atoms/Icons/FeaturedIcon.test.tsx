import React from "react";
import { render, screen } from "@testing-library/react";
import { FeaturedIcon } from "./FeaturedIcon";
import { IconPlus } from "../Icons/Icons";

describe("FeaturedIcon", () => {
    test('renders featured icon component correctly and has all props', () => {
        const container1 = render(
            <FeaturedIcon
            size="xl"
            type="primary"
            icon={<IconPlus size='20' color='stroke-gray-700 dark:stroke-white' fill='' />}
            />
        );
        expect(container1).toMatchSnapshot();
        
        const container2 = render(
            <FeaturedIcon
            size="md"
            type="gray"
            icon={<IconPlus size='20' color='stroke-gray-700 dark:stroke-white' fill='' />}
            />
        );
        expect(container2).toMatchSnapshot();

        const container3 = render(
            <FeaturedIcon
            size="lg"
            type="error"
            icon={<IconPlus size='20' color='stroke-gray-700 dark:stroke-white' fill='' />}
            />
        );
        expect(container3).toMatchSnapshot();
    
        const container4 = render(
            <FeaturedIcon
            size="sm"
            type="success"
            icon={<IconPlus size='20' color='stroke-gray-700 dark:stroke-white' fill='' />}
            />
        );
        expect(container4).toMatchSnapshot();
    });
});
