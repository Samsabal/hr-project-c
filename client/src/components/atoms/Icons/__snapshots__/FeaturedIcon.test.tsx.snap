// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedIcon renders featured icon component correctly and has all props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="p-3 border-8 bg-primary-100 border-primary-50 dark:border-dark-700 dark:bg-dark-600
            flex items-center justify-center rounded-full box-border"
      >
        <svg
          class="stroke-gray-700 dark:stroke-white"
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
        >
          <path
            d="M12 5V19M5 12H19"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="p-3 border-8 bg-primary-100 border-primary-50 dark:border-dark-700 dark:bg-dark-600
            flex items-center justify-center rounded-full box-border"
    >
      <svg
        class="stroke-gray-700 dark:stroke-white"
        fill="none"
        height="20"
        viewBox="0 0 24 24"
        width="20"
      >
        <path
          d="M12 5V19M5 12H19"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FeaturedIcon renders featured icon component correctly and has all props 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="p-3 border-8 bg-primary-100 border-primary-50 dark:border-dark-700 dark:bg-dark-600
            flex items-center justify-center rounded-full box-border"
      >
        <svg
          class="stroke-gray-700 dark:stroke-white"
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
        >
          <path
            d="M12 5V19M5 12H19"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
    <div>
      <div
        class="p-2 border-8 bg-gray-100 border-gray-50 dark:border-dark-700 dark:bg-dark-600
            flex items-center justify-center rounded-full box-border"
      >
        <svg
          class="stroke-gray-700 dark:stroke-white"
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
        >
          <path
            d="M12 5V19M5 12H19"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="p-2 border-8 bg-gray-100 border-gray-50 dark:border-dark-700 dark:bg-dark-600
            flex items-center justify-center rounded-full box-border"
    >
      <svg
        class="stroke-gray-700 dark:stroke-white"
        fill="none"
        height="20"
        viewBox="0 0 24 24"
        width="20"
      >
        <path
          d="M12 5V19M5 12H19"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FeaturedIcon renders featured icon component correctly and has all props 3`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="p-3 border-8 bg-primary-100 border-primary-50 dark:border-dark-700 dark:bg-dark-600
            flex items-center justify-center rounded-full box-border"
      >
        <svg
          class="stroke-gray-700 dark:stroke-white"
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
        >
          <path
            d="M12 5V19M5 12H19"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
    <div>
      <div
        class="p-2 border-8 bg-gray-100 border-gray-50 dark:border-dark-700 dark:bg-dark-600
            flex items-center justify-center rounded-full box-border"
      >
        <svg
          class="stroke-gray-700 dark:stroke-white"
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
        >
          <path
            d="M12 5V19M5 12H19"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
    <div>
      <div
        class="p-2.5 border-8 bg-error-100 border-error-50 dark:border-dark-700 dark:bg-dark-600
            flex items-center justify-center rounded-full box-border"
      >
        <svg
          class="stroke-gray-700 dark:stroke-white"
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
        >
          <path
            d="M12 5V19M5 12H19"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="p-2.5 border-8 bg-error-100 border-error-50 dark:border-dark-700 dark:bg-dark-600
            flex items-center justify-center rounded-full box-border"
    >
      <svg
        class="stroke-gray-700 dark:stroke-white"
        fill="none"
        height="20"
        viewBox="0 0 24 24"
        width="20"
      >
        <path
          d="M12 5V19M5 12H19"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FeaturedIcon renders featured icon component correctly and has all props 4`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="p-3 border-8 bg-primary-100 border-primary-50 dark:border-dark-700 dark:bg-dark-600
            flex items-center justify-center rounded-full box-border"
      >
        <svg
          class="stroke-gray-700 dark:stroke-white"
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
        >
          <path
            d="M12 5V19M5 12H19"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
    <div>
      <div
        class="p-2 border-8 bg-gray-100 border-gray-50 dark:border-dark-700 dark:bg-dark-600
            flex items-center justify-center rounded-full box-border"
      >
        <svg
          class="stroke-gray-700 dark:stroke-white"
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
        >
          <path
            d="M12 5V19M5 12H19"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
    <div>
      <div
        class="p-2.5 border-8 bg-error-100 border-error-50 dark:border-dark-700 dark:bg-dark-600
            flex items-center justify-center rounded-full box-border"
      >
        <svg
          class="stroke-gray-700 dark:stroke-white"
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
        >
          <path
            d="M12 5V19M5 12H19"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
    <div>
      <div
        class="p-1.5 border-8 bg-success-100 border-success-50 dark:border-dark-700 dark:bg-dark-600
            flex items-center justify-center rounded-full box-border"
      >
        <svg
          class="stroke-gray-700 dark:stroke-white"
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
        >
          <path
            d="M12 5V19M5 12H19"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="p-1.5 border-8 bg-success-100 border-success-50 dark:border-dark-700 dark:bg-dark-600
            flex items-center justify-center rounded-full box-border"
    >
      <svg
        class="stroke-gray-700 dark:stroke-white"
        fill="none"
        height="20"
        viewBox="0 0 24 24"
        width="20"
      >
        <path
          d="M12 5V19M5 12H19"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
